app/controllers/income/estimatedPay/update/IncomeUpdateWorkingHoursController.scala

```diff
@@ -43,20 +43,17 @@ class IncomeUpdateWorkingHoursController @Inject()(
   def workingHoursPage: Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
     implicit val user: AuthedUser = request.taiUser
 
-    for {
-      incomeSourceEither <- IncomeSource.create(journeyCacheService)
-      workingHours       <- journeyCacheService.currentValue(UpdateIncome_WorkingHoursKey)
-    } yield {
-
-      incomeSourceEither match {
-        case Right(incomeSource) =>
-          Ok(
-            workingHoursView(
-              HoursWorkedForm.createForm().fill(HoursWorkedForm(workingHours)),
-              incomeSource.id,
-              incomeSource.name))
-        case Left(_) => Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())
-      }
+    (IncomeSource.create(journeyCacheService), journeyCacheService.currentValue(UpdateIncome_WorkingHoursKey)).mapN {
+      case (incomeSourceEither, workingHours) =>
```

Comment:
u could do: 

`case (Right(incomeSource), workingHours) => 
    Ok(
              workingHoursView(
                HoursWorkedForm.createForm().fill(HoursWorkedForm(workingHours)),
                incomeSource.id,
                incomeSource.name))
case (Left(_), _) =>    Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad())             
`

Reply:
the second case could also be `case _ => `